lib/core/engine.js

```diff
@@ -60,18 +62,50 @@ class Engine {
 
     this.options = merge({}, defaults, options);
     this.runDelegate = engineDelegate.createDelegate(this.options);
+    this.xvfb = this.setupXvfb();
+  }
+
+  setupXvfb() {
+    const width = get(this.options, 'chrome.mobileEmulation.width');
+    const height = get(this.options, 'chrome.mobileEmulation.height');
+
+    let widthAndHeight = (width && height) ? width + 'x' + height : this.options.viewPort || '1500x1200';
+
+    const xvfb = new Xvfb({
+        displayNum: '99',
+        silent: false,
+        reuse: true,
+        xvfb_args: ['-ac', '-nolisten', 'tcp', '-screen', 0, widthAndHeight + 'x16']
+    });
+
+    return Promise.promisifyAll(xvfb);
+  }
+
+
+  startXvfb() {
+    if (this.options.xvfb) {
+        return this.xvfb.startAsync();
+    }
+    else return Promise.resolve();
+  }
+
+  stopXvfb() {
+    if (this.xvfb) {
+      return this.xvfb.stopAsync();
+    }
+    else return Promise.resolve();
   }
 
   start() {
     const runDelegate = this.runDelegate;
       if (this.options.connectivity && this.options.connectivity.profile !== 'native' && this.options.connectivity.profile !== undefined) {
-    return connectivity.set(this.options).then(function() {
+    return this.startXvfb().then(connectivity.set(this.options)).then(function() {
```

Comment:
then takes a function, not a promise, as argument. So this should be:

return this.startXvfb()
           .then(() => connectivity.set(this.options))
           .then(function() {...